core/src/main/java/bisq/core/trade/TradeUtil.java

```diff
@@ -173,6 +174,27 @@ public String getPaymentMethodNameWithCountryCode(Trade trade) {
         return paymentMethodDescription;
     }
 
+    /**
+     * Returns a string describing a trader's role for a given trade.
+     * @param trade Trade
+     * @return String describing a trader's role for a given trade
+     */
+    public String getRole(Trade trade) {
+        Contract contract = trade.getContract();
+        if (contract == null)
+            throw new IllegalStateException(format("could not get role because no contract was found for trade '%s'",
+                    trade.getShortId()));
+
+        Offer offer = trade.getOffer();
+        if (offer == null)
+            throw new IllegalStateException(format("could not get role because no offer was found for trade '%s'",
+                    trade.getShortId()));
+
+        return getRole(contract.isBuyerMakerAndSellerTaker(),
+                offer.isMyOffer(keyRing),
+                offer.getCurrencyCode());
```

Comment:
I don't get why currenycode should be needed to decide role

Reply:
Below is an example of the console output for a CLI's `gettrade id` command, where the role = `BTC buyer as taker`, showing what is being bought or sold.

```
ID      My Role             Price in USD for 1 BTC  Amount(BTC)  Tx Fee(BTC)  Taker Fee(BTC)  Deposit Published  Deposit Confirmed  Fiat Sent  Fiat Received  Payout Published  Withdrawn  
AGnSOw  BTC buyer as taker             13,263.6578   0.12500000   0.00068800      0.00037500  YES                NO                 NO         NO             NO                NO        
```

Reply:
Ok, I get it, thanks.